src/openrct2/actions/SignSetStyleAction.hpp

```diff
@@ -77,23 +77,9 @@ DEFINE_GAME_ACTION(SignSetStyleAction, GAME_COMMAND_SET_SIGN_STYLE, GameActionRe
         }
         else
         {
-            TileElement* tileElement = map_get_first_element_at(coords.x / 32, coords.y / 32);
-            bool wallFound = false;
-            do
-            {
-                if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
-                    continue;
-
-                rct_scenery_entry* scenery_entry = tileElement->AsWall()->GetEntry();
-                if (scenery_entry->wall.scrolling_mode == SCROLLING_MODE_NONE)
-                    continue;
-                if (tileElement->AsWall()->GetBannerIndex() != (BannerIndex)_bannerIndex)
-                    continue;
-                wallFound = true;
-                break;
-            } while (!(tileElement++)->IsLastForTile());
-
-            if (!wallFound == false)
+            WallElement* wallElement = banner_get_scrolling_wall_tile_element((BannerIndex)_bannerIndex);
```

Comment:
try to prefer using the `static_cast<BannerIndex>(` style. Its not the end of the world to use C style cast but `xxx_casts` are clearer as to what you are trying to achieve.

Reply:
OK I've changed that now :)